src/promptflow/promptflow/core/_flow.py

```diff
@@ -167,3 +183,197 @@ async def invoke(self, inputs: dict, *, connections: dict = None, **kwargs) -> "
+
+    @classmethod
+    def _load(cls, path: Path, **kwargs):
+        return cls(path=path, **kwargs)
+
+    # region overrides
+    @classmethod
+    def load(
+        cls,
+        source: Union[str, PathLike],
+        raise_error=True,
+        **kwargs,
+    ) -> "Prompty":
+        """
+        Direct load non-dag flow from prompty file.
+
+        :param source: The local prompt file. Must be a path to a local file.
+            If the source is a path, it will be open and read.
+            An exception is raised if the file does not exist.
+        :type source: Union[PathLike, str]
+        :param raise_error: Argument for non-dag flow raise validation error on unknown fields.
+        :type raise_error: bool
+        :return: A Prompty object
+        :rtype: Prompty
+        """
+        source_path = Path(source)
+        if not source_path.exists():
+            raise UserErrorException(f"Source {source_path.absolute().as_posix()} does not exist")
+
+        if source_path.suffix != PROMPTY_EXTENSION:
+            raise UserErrorException("Source must be a file with .prompty extension.")
+        return cls._load(path=source_path, **kwargs)
+
+    @staticmethod
+    def _parse_prompty(path):
+        """ """
+        with open(path, "r", encoding=DEFAULT_ENCODING) as f:
+            prompty_content = f.read()
+        pattern = r"-{3,}\n(.*)-{3,}\n(.*)"
+        result = re.search(pattern, prompty_content, re.DOTALL)
+        if not result:
+            raise UserErrorException(
+                "Illegal formatting of prompty. The prompt file is in markdown format and can be divided into two "
+                "parts, the first part is in YAML format and contains connection and model information. The second "
+                "part is the prompt template."
+            )
+        config_content, prompt_template = result.groups()
+        configs = load_yaml_string(config_content)
+        return configs, prompt_template
+
+    def _resolve_inputs(self, inputs):
+        resolved_inputs = {}
+        for k, v in inputs.items():
+            if isinstance(v, dict):
+                resolved_inputs[k] = v
+            else:
+                resolved_inputs[k] = {"type": ValueType.from_value(v).value, "default": v}
+        return resolved_inputs
+
+    def _validate_inputs(self, input_values):
+        resolved_inputs = {}
+        missing_inputs = []
+        for input_name, value in self.inputs.items():
+            if input_name not in input_values and "default" not in value:
+                missing_inputs.append(input_name)
+                continue
+            resolved_inputs[input_name] = input_values.get(input_name, value.get("default", None))
+        if missing_inputs:
+            raise MissingRequiredInputError(f"Missing required inputs: {missing_inputs}")
+        return resolved_inputs
+
+    def __call__(self, *args, **kwargs):
```

Comment:
what's current trace tree now? share an example?

Reply:
Improve trace in next PR